File: bot/src/main/java/com/ryderbelserion/discord/api/commands/interfaces/CommandFlow.java
package com.ryderbelserion.discord.api.commands.interfaces;

import com.ryderbelserion.discord.api.commands.CommandEngine;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;
import org.jspecify.annotations.NonNull;
import java.util.List;

public interface CommandFlow {

    void addCommands(@NonNull final List<CommandEngine> commands);

    void addStaticGuildCommands(@NonNull final Guild guild, @NonNull final List<CommandEngine> commands);

    void addGuildCommands(@NonNull final Guild guild, @NonNull final List<CommandData> commands);

    void purgeGuildCommands(@NonNull final Guild guild);

    void purgeCommands();
}

File: src/main/kotlin/com/ryderbelserion/api/builders/commands/interfaces/CommandFlow.kt
package com.ryderbelserion.api.builders.commands.interfaces

import com.ryderbelserion.api.builders.commands.CommandEngine
import net.dv8tion.jda.api.entities.Guild
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.OptionData

interface CommandFlow {

    fun addCommand(engine: CommandEngine)

    fun addCommand(engine: CommandEngine, optionData: OptionData)

    fun addCommand(engine: CommandEngine, optionDataList: MutableList<OptionData?>)

    fun addCommand(engine: CommandEngine, type: OptionType, name: String, description: String)

    fun addGuildCommand(guild: Guild, engine: CommandEngine)

    fun addGuildCommand(guild: Guild, engine: CommandEngine, optionData: OptionData)

    fun addGuildCommand(guild: Guild, engine: CommandEngine, optionDataList: MutableList<OptionData?>)

    fun addGuildCommand(guild: Guild, engine: CommandEngine, type: OptionType, name: String, description: String)

    fun addGuildCommands(guild: Guild, vararg engines: CommandEngine)

    fun addCommands(vararg engines: CommandEngine)

    fun purgeGuildCommands(guild: Guild)

    fun purgeGlobalCommands()

}